pkg/parser/expression/test.tmpl

```diff
@@ -0,0 +1,116 @@
+package expression
+
+import (
+    "fmt"
+    "go/ast"
+    "go/parser"
+    "testing"
+
+//    "github.com/gofed/symbols-extractor/pkg/parser/alloctable"
+//    "github.com/gofed/symbols-extractor/pkg/parser/symboltable"
+    "github.com/gofed/symbols-extractor/pkg/parser/types"
+
+//    typeparser "github.com/gofed/symbols-extractor/pkg/parser/type"
+      gotypes    "github.com/gofed/symbols-extractor/pkg/types"
+)
+
+const gopkg string = "github.com/gofed/symbols-extractor/pkg/parser/testdata/valid"
+const gocode string = `
+package exprtest
+{{range .SymTab}}
+  {{.}}
+{{end}}
+`
+
+{{define "nlspace"}}
+
+{{end}}
+
+{{/* 
+# The skeleton for each target:
+#  {{if eq $target "<set-target-you-want-to-use-here>"}}
+#    {{range $i, $elem := .Tests}}func Test{{tittle $target}}{{$i}} (t *testing.T) {
+#      // PUT YOUR CODE HERE
+#   }{{template "nlspace"}}{{end}}{{end}}
+*/}}
+
+func initST() (*types.Config, error) {
+   config := prepareParser(gopkg)
+   astF, _, err := getAst(gopkg, "", gocode)
+   if err != nil {
+     return nil,  fmt.Errorf("Broken test! Fix test suite: %v", err)
+   }
+   err = parseNonFunc(config, astF)
+   if err != nil {
+     return nil, err
+   }
+
+   return config, nil
+}
+
+func initExprTest(expr_str string) (*types.Config, ast.Expr, error) {
+        config, errT := initST()
+        if errT != nil {
+          return nil, nil, errT
+        }
+
+        expr, errE := parser.ParseExpr(expr_str)
+        if errE != nil {
+          return nil, nil, fmt.Errorf("Broken test! Fix test suite: %v", errE)
+        }
+
+        return config, expr, nil
+}
+
+{{range .Targets}}
+  {{$target := .Target}}
+
+  {{if eq $target "parseBinaryExpr"}}{{range $i, $elem := .Tests}}
+    func Test{{tittle $target}}{{$i}} (t *testing.T) {
+        expr_str := "{{.Code}}"
+        expected_res := &gotypes.{{BuiltinOrIdent .Expected}}{{"{Def:\""}}{{.Expected}}{{"\"}"}}
+        config, expr, errE := initExprTest(expr_str)
+        if errE != nil {
+          t.Errorf("%v", errE)
+          return
+        }
+
+        resi, err := config.ExprParser.(*Parser).parseBinaryExpr(expr.(*ast.BinaryExpr))
+        if err != nil {
+          msgf := "Unexpected error for expr '%s': %v\n"
+          t.Errorf(msgf, expr_str, err)
```

Comment:
You can put the `msgf` right into the `t.Errorf`. Given the `msgf` is short, it will fit in one line and still look good.